### PR TITLE
Fix 3007

### PR DESCRIPTION
**Pull request by amedora**
## Description

Not only for mermaid, other code block like 'sequence' as below has potentially XSS problem.
```
```sequence
<iframe src="javascript:alert('xss');">
\```
```

## Issue fixed

#3007 

## Type of changes

- :radio_button: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :white_circle: All existing tests have been passed
- :white_circle: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

File: browser/components/MarkdownPreview.js [hunk 1]
```diff
@@ -845,7 +845,7 @@ export default class MarkdownPreview extends React.Component {
     _.forEach(
       this.refs.root.contentWindow.document.querySelectorAll('.mermaid'),
       el => {
-        mermaidRender(el, htmlTextHelper.decodeEntities(el.innerHTML), theme)
+        mermaidRender(el, htmlTextHelper.encodeEntities(el.innerHTML), theme)
```
Comment (ZeroX-DG) [inline browser/components/MarkdownPreview.js]: Your code fixed the XSS but the mermaid is unusable after your fix.
Example:
```
```mermaid
sequenceDiagram
loop every day
    Alice->>John: Hello John, how are you?
    John-->>Alice: Great!
end
```

The encodeEntities encoded the content of the mermaid code so it becomes invalid. It seems encodeEntities is not a good way to fix this bug. @Rokt33r what do you suggest for this? Maybe a write a plugin to parse mermaid directly from markdown?

Comment (AWolf81) [reply browser/components/MarkdownPreview.js]: Why is this change needed? I think the XSS is also working with-out this change - tested locally with just the `_fence` change in `markdown-it-sanitize-html.js`. 
Tested with the following [gist](https://gist.github.com/rebirthwyw/9836b2445abd4c1358e3d68ff2415247).

Comment (amedora) [reply browser/components/MarkdownPreview.js]: > Your code fixed the XSS but the mermaid is unusable after your fix.
oh, nice catch. I confirmed that too.

@AWolf81 
Even if we sanitized a content in markdown-it-sanitize-html.js, it's going to be original in MarkdownPreview.js:715
```js
    const renderedHTML = this.markdown.render(value)
    attachmentManagement.migrateAttachments(value, storagePath, noteKey)
    this.refs.root.contentWindow.document.body.innerHTML = attachmentManagement.fixLocalURLS(
      renderedHTML,
      storagePath
    )
```
I think we have to do something for mermaid only (other plugins don't have this vulnerability).

**Comment by amedora** [comment]
I think that most straightforward way to fix this is passing "flowchart: { htmlLabels: false }" option to mermaidAPI.initialize().
https://github.com/BoostIO/Boostnote/blob/76335f78aca94b39841b03af4c212bec65862e5b/browser/components/render/MermaidRender.js#L29-L33
However it would be drawback for users who use HTML in mermaid charts.

**Comment by AWolf81** [comment]
@amedora maybe we can add this as option - disabled html labels by default. With a warning at the option that there is a risk of xss. So Mermaid html label users can decide and accept the risk.

That's not perfect but it will fix xss. Maybe we can open an issue at Mermaid so we can track if there will be a fix.

**Comment by amedora** [comment]
It's now configurable in the option panel. works perfect on my end.
option

**Comment by ZeroX-DG** [comment]
LGTM :tada: But I'll need review from @Rokt33r before I can approve this.

**Pull request merged**